src/modules/samples/rust/src/lib.rs

```diff
@@ -94,7 +96,37 @@ pub extern "C" fn MmiSet(
     payload: MmiJsonString,
     payload_size_bytes: c_int,
 ) -> c_int {
-    unimplemented!("MmiSet is not yet implemented");
+    if client_session.is_null() {
+        println!("MmiSet called with null clientSession");
+        EINVAL
+    } else {
+        let sample: &mut Sample = unsafe { &mut *(client_session as *mut Sample) };
+        let component_name: &CStr = unsafe { CStr::from_ptr(component_name) };
+        let object_name: &CStr = unsafe { CStr::from_ptr(object_name) };
+        let payload: &CFixedStr = unsafe { CFixedStr::from_ptr(payload, payload_size_bytes as usize) };
```

Comment:
Here and elsewhere, does it make sense to wrap these 4 lines in an `unsafe { }` block? What is the benefit to using unsafe for each line? If there is no difference, it might be clearer to group these things together in one block.

Reply:
I'd argue this better. If we did one big block, we'd have to place safe functions (like mmi_set_helper) in the unsafe block because of scoping: all of the variables would be in the unsafe block scope so the helper method would have to be in the block to access them.